### ISSUE TITLE
Add a camera to debug draw classes

### ISSUE THREAD

**Issue by viblo** [viblo/pymunk#106]
It would be convenient if it was possible to pan and zoom around. Having a camera would be a nice addition and make the debug drawing more useful

**Comment by Berserker66** [comment]
I might be getting on that. Still in the deciding/planning phase. I could use it for my project.

Things I'd like to do:
Have a pluggable "camera" function, that takes in the normal render information, mostly coordinates and returns the changed data. That way you can have a dummy function, just returning the old data for projects without camera and more complex projects can add stuff like y-flip to their camera function.

Secondly, I'd like a "raw data export" draw, that just exports the data in something like python array.array objects, so they can be logged to a file and later reviewed, or be plugged into a renderer.

**Comment by viblo** [comment]
Just a small note: its possible to pickle the state of most things. So if  you want to save to file you can already do that. However, it probably contains more information than needed for just render, and its not really optimized for quick render. 

There is also the base SpaceDebugDrawOptions class that "draws" the space by printing it. It could easily be modified to instead store the information in some structure.

(On the other hand, if you are making a more complex program maybe you need customized drawing/exporting to make it flexible and fast, the debug draw and pickle are not really made for that)

**Comment by Berserker66** [comment]
In my project I intend to use it as an optional debug draw. The main renderer will not use this method. But it could be useful to overlay the raw physics information, since what you see in regular rendering can very much differ from the physics- for example you can tell if a sprite and its polygon don't match up well and other such things, that aren't useful for actual play and are fast enough if it's just a few fps, since it's just for debuggin.

**Comment by viblo** [comment]
Revisiting this because of https://stackoverflow.com/questions/68540117/scale-graphics-with-pygame-util

After some further thinking, I think I could add a transform property to the class `SpaceDebugDrawOptions` class, and then a user could do like this to scale, or construct any other kind of transform they need:
```
options = SpaceDebugDrawOptions() # one of the specific implementations like pygame_util.DrawOptions
...
options.transform = Transform.scaling(2)
space.debug_draw(options)
```

**Comment by viblo** [comment]
Fixed in Pymunk 6.1 by the new transform property on SpaceDebugDrawOptions.
